packages/material-ui-styles/src/index.d.ts

```diff
@@ -307,6 +307,18 @@ declare module '@material-ui/styles/withStyles' {
 declare module '@material-ui/styles/withTheme' {
   import { PropInjector } from '@material-ui/core';
 
+  export interface WithThemeCreatorOption<Theme> {
+    defaultTheme?: Theme;
+  }
+
+  export type WithThemeCreator<Theme> = (
+    option?: WithThemeCreatorOption<Theme>,
+  ) => PropInjector<WithTheme<Theme>, Partial<WithTheme<Theme>>>;
+
+  export function withThemeCreator<Theme>(
+    option?: WithThemeCreatorOption<Theme>,
+  ): WithThemeCreator<Theme>;
```

Comment:
I'm not sure to read this correctly. Does it mean?
`withThemeCreator(options) => WithThemeCreator(options) => the higher order component`
I'm don't really know TypeScript.

Reply:
I think I did a mistake. Indeed.

I was fooled by the definition of withTheme
```tsx
export default function withTheme<Theme>(): PropInjector<
    WithTheme<Theme>,
    Partial<WithTheme<Theme>>
  >;
```

When the breaking change was made, the typing was not updated
```diff
-const DeepChild = withTheme()(DeepChildRaw);
+const DeepChild = withTheme(DeepChildRaw);
```